### PR TITLE
Ported to Gulp

### LINKED ISSUE #21: [Proposal] Use gulp to build instead of make
We need this for our own purposes, which would include gulp watch to automatically build. Right now there is no mention in Readme on how to build.

We'll PR against this issue. Please leave open.

### PR DESCRIPTION
**Pull request by tintou**
Tested the resulting files in ethercalc.
Also using a small html file showing a single socialcalc page
fixes #21

### CODE REVIEW

**Comment by marcelklehr** [comment]
Wow, nice work!

File: README.md [hunk 1]
```diff
@@ -18,6 +18,15 @@ socialCalcControl.InitializeSpreadsheetControl(el /*, height, width, spacebelow*
 
 This package also works in node.js. You don't need to callc `InitializeSpreadsheetControl` here, because that is just initializing the rendering.
 
+## Build from sources
+
+SocialCalc uses npm and gulp, process as followed:
+```bash
+npm install
```
Comment (marcelklehr) [inline README.md]: Will this also work, when socialcalc is used as a dependency? (Because the gulp deps are devDeps...)

**Comment by tintou** [comment]
@marcelklehr You're right, better do it the safe way, pushed a fix

**Comment by eddyparkinson** [comment]
Thanks for the contribution. 
@tintou Please add a summary of what has been tested. And I will merge it in. Thanks.
@marcelklehr Are you happy with it?

**Comment by tintou** [comment]
I've tested it in ethercalc with the french locale, everything seems to work as expected.

**Comment by eddyparkinson** [comment]
Do we remove the make file? As I assume we would need to change the make file and the gulp file if we change how SocialCalc.js is created.  I assume both would need to be maintained.

Related issue -  I change my make file to 
   copy SocialCalc.js   to node_modules/socialcalc/SocialCalc.js 
So that I can see the change in ethercalc. Does it have an easy way to do that?

**Comment by tintou** [comment]
@eddyparkinson Do you clone socialcalc in the same directory as ethercalc ? I would suggest you work in node_modules/socialcalc/ directly so that it's always up-to-date

**Comment by tintou** [comment]
I'm not opposed to the removal of the makefile

**Comment by eddyparkinson** [comment]
@tintou Please remove the make file or change the make file so it uses gulp.  Just to avoid errors caused by looking after a duplicate.

**Comment by tintou** [comment]
@eddyparkinson Makefile deleted :slightly_smiling_face:

**Review by eddyparkinson — APPROVED** [review]
Looks correct - Would be best to delete the make file. 

will merge when I have some time.

**Pull request merged**